src/common/vt_dsp/vt_dsp_endian.h

```diff
@@ -6,88 +6,41 @@
 #define vt_read_int16BE vt_write_int16BE
 #define vt_read_float32LE vt_write_float32LE
 
-__forceinline int swap_endian(int t)
-{
-#if (__linux__ || MAC || _M_X64)
-   return ((t << 24) & 0xff000000) | ((t << 8) & 0x00ff0000) | ((t >> 8) & 0x0000ff00) |
-          ((t >> 24) & 0x000000ff);
-#else
-   // GCC does not seem to respect that eax must be correct when it returns. Add: ret eax?
-   __asm
-   {
-		mov		eax, t
-		bswap	eax
-   }
-#endif
-}
-
 __forceinline int vt_write_int32LE(int t)
 {
-#if PPC
-   return swap_endian(t);
-#else
    return t;
-#endif
 }
 
 __forceinline float vt_write_float32LE(float f)
 {
-#if PPC
-   int t = *((int*)&f);
-   t = swap_endian(t);
-   return *((float*)&t);
-#else
    return f;
-#endif
 }
 
 __forceinline int vt_write_int32BE(int t)
 {
-#if !PPC
-   return swap_endian(t);
+#if (__linux__ || MAC || _M_X64)
+   return __builtin_bswap32(t);
 #else
-   return t;
```

Comment:
This seems like it’s an actually diff, not just a PPC removal. Not super clear to me from inspection it is the same. Seems the old behaviour was just “return t” and now we have two branches of bswap32.

Reply:
Hold on, checking...

Reply:
Yeah, well there is only one call site for `swap_endian()` after the PPC code has been removed so I open coded it to `vt_write_int32BE()` and use compiler builtins instead of just copy-pasting the original code. Should probably mention this detail in the commit message.